extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -0,0 +1,606 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Action,
+  StateProperty,
+  UIType,
+  getStoreService,
+} from './amp-story-store-service';
+import {CSS} from '../../../build/amp-story-tooltip-1.0.css';
+import {EventType, dispatch} from './events';
+import {Services} from '../../../src/services';
+import {addAttributesToElement, closest, matches} from '../../../src/dom';
+import {createShadowRootWithStyle, getSourceOriginForElement} from './utils';
+import {dev, devAssert, user} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {getAmpdoc} from '../../../src/service';
+import {htmlFor, htmlRefs} from '../../../src/static-template';
+import {isProtocolValid, parseUrlDeprecated} from '../../../src/url';
+import {setImportantStyles, toggle} from '../../../src/style';
+
+/**
+ * Enum of elements that can be expanded.
+ * @enum {string}
+ */
+const EXPANDABLE_COMPONENTS = {
+  TWITTER: 'amp-twitter',
+};
+
+/**
+ * Enum containing all interactive components.
+ * @enum {string}
+ */
+const interactiveComponents = Object.assign({}, EXPANDABLE_COMPONENTS, {
+  EXPANDED_VIEW_OVERLAY: '.i-amphtml-story-expanded-view-overflow, ' +
+    '.i-amphtml-expanded-view-close-button',
+  LINK: 'a[href]',
+});
+
+/**
+ * Selectors that should delegate to AmpStoryEmbeddedComponent.
+ */
+export function embeddedComponentSelectors() {
+  // Using indirect invocation to prevent no-export-side-effect issue.
+  return interactiveComponents;
+}
+
+/**
+ * Builds expanded view overlay for expandable components.
+ * @param {!Element} element
+ * @return {!Element}
+ */
+const buildExpandedViewOverlay = element => htmlFor(element)`
+    <div class="i-amphtml-story-expanded-view-overflow
+        i-amphtml-story-system-reset">
+      <span class="i-amphtml-expanded-view-close-button" role="button">
+        &times;
```

Comment:
I super recently found out that we should actually use a SVG for that, since the material icon is a bit thinner. Could you try using this one and see how it looks? Not a big deal and feel free to skip this comment if it introduces too much complexity.

`<svg xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 36 36" fill="rgba(0, 0, 0, 0.54)"><path d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"/><path d="M0 0h36v36H0z" fill="none"/></svg>`

Reply:
Done